tutorials/5-docker-intro/index.md

```diff
@@ -0,0 +1,288 @@
+
+```sh
+$ docker run -it node:lts-slim
+```
+
+`-it` is short for `--interactive --tty` and means with input and output instead of a fully detached container. Learn more with `docker run --help`. This should return you a Node prompt:
+
+```javascript
+Welcome to Node.js v18.12.1.
+Type ".help" for more information.
+>
+```
+
+Version 18 is a long-term support (`lts`) one. If you type `.help` it tells you what you need. Exit with `.exit`.
+
+That was fast. What happened? Docker downloaded this image from the hub, and launched it. By default the image is configured so that the container launches `node`, which is what you got. You did not do much interesting, though.
+
+### Open a shell
+
+What if you want to connect to your container with a shell? After all, this is running Linux. Because this image defines Node as its [**entry point**](https://hub.docker.com/layers/library/node/lts-slim/images/sha256-3139aa3e8915e7c135623498d29f20a75ee3bfc41cf321ceaa59470b2fffc1a5?context=explore), you need to override it:
+
+```sh
+$ docker run -it --entrypoint bash node:lts-slim
+```
+
+Now you see something different:
+
+```sh
+root@bd3982cf3d68:/#
+```
+
+You are `root`! But only in the container, not in the Linux host. Typically, programs running in a container are left running with `root` as the container takes care of the isolation.
+
+The image `node:lts-slim`, as the `slim` part lets on, does not have much else beside Node. Try:
+
+```sh
+$ curl
+```
+
+It should tell you:
+
+```txt
+bash: curl: command not found
+```
+
+This means that you need to pick your image carefully and even sometimes you will also have to install the tools you need. Exit with a regular `exit`.
+
+### Your own image
+
+Suppose you need a Node container with the `curl` command available. For that, you need to **build** your own image. It so happens that `node:lts-slim` is built on Debian, so you can use `apt-get` to install new programs.
+
+Create a new file named `Dockerfile` with:
+
+```Dockerfile
+FROM node:lts-slim
+
+RUN apt-get update
+RUN apt-get install -y curl
+```
+
+You have to `update` the package registry because the image is kept slim by not even having a local copy. Build your new image with a name of your choosing:
+
+```sh
+$ docker build . -t node-with-curl:lts-slim
+```
+
+Now your image is ready. Can you call up `curl`? Type:
+
+```sh
+$ docker run -it --entrypoint curl node-with-curl:lts-slim
+```
+
+It should return:
+
+```txt
+Use "--help category" to get an overview of all categories.
+For all options use the manual or "--help all".
+```
+
+So yes, you now have `curl` and Node on the same container.
+
+### Hello World
+
+What if you wanted to use Node to print `Hello World`? Create the Javascript file that can do that:
+
+```sh
+$ echo "console.log(\"Hello World\");" > test.js
+```
+
+Now pass it to your container:
+
+```sh
+$ docker run -i node:lts-slim test.js
```

Comment:
@xavierlepretre - apologies for stepping outside of language review territory! But on this line, is it possible that **-i** should be **-it**?

I ask because the instructions/changes for lines 162-167 don't specify precisely what is being changed, and I see two differences; whereas 176-179 does specify one precise change, removing the "-t".

Reply:
Yep, please put the `t`. From my testing, there is no difference. Not sure why there was none here.